ra/registration-authority.go

```diff
@@ -97,9 +97,10 @@ func validateEmail(address string, resolver bdns.DNSResolver) (prob *probs.Probl
 	var resultMX []string
 	var resultA []net.IP
 	resultMX, err = resolver.LookupMX(domain)
-
+	recQ := "MX"
 	if err == nil && len(resultMX) == 0 {
 		resultA, err = resolver.LookupHost(domain)
+		recQ = "A"
```

Comment:
The `recQ` thing here is kind of kludgey, but it's abundantly clear to me what's going on. The alternatives of code duplication, or making a helper method, do seem worse.